### PR TITLE
feat(crypto): Add customized event type for the forwarded room key

### PR DESCRIPTION
**Pull request by poljar**

### CODE REVIEW

File: crates/matrix-sdk-crypto/src/olm/group_sessions/mod.rs [hunk 1]
```diff
@@ -123,16 +121,19 @@ impl TryInto<ToDeviceForwardedRoomKeyEventContent> for ExportedRoomKey {
...
-            }
-            .into())
+            Ok(ForwardedRoomKeyContent::MegolmV1AesSha2(
+                ForwardedMegolmV1AesSha2Content {
+                    room_id: self.room_id,
+                    session_id: self.session_id,
+                    session_key: self.session_key,
+                    claimed_sender_key: self.sender_key,
+                    claimed_ed25519_key: Ed25519PublicKey::from_base64(claimed_key)
+                        .map_err(|_| ())?,
```
[the remark this answers is not in the document]
Comment (poljar) [reply crates/matrix-sdk-crypto/src/olm/group_sessions/mod.rs:131]: Yeah, let's do that, even though the new one might eventually go away once we deserialize into the `Ed25519Publickey` type.

[the remark this answers is not in the document]
Comment (poljar) [reply crates/matrix-sdk-crypto/src/olm/group_sessions/mod.rs]: This turned to be a bit of a bigger change, but now we have a proper error for this: https://github.com/matrix-org/matrix-rust-sdk/pull/920/commits/2943a86f3ad6465dc4084ff88d77d7ed7315292b.

**Review by jplatte — APPROVED** [review]
Commit message has a typo: "exporing" instead of "exporting" (could actually also be read as a typo of "expiring" 😄)

Otherwise LGTM!

**Comment by poljar** [comment]
> Commit message has a typo: "exporing" instead of "exporting" (could actually also be read as a typo of "expiring" smile)
> 
> Otherwise LGTM!

Gah, thanks. Fixed in a force push, got rid of the merge conflict as well, and somehow resurrected the CI.

**Pull request merged**